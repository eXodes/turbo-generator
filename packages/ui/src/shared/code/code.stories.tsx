import type { Meta, StoryObj } from "@storybook/react";

import { Code } from "./code";

const meta = {
  title: "Shared/Code",
  component: Code,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Code>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: `import { Code } from "@components/code"`,
  },
};
