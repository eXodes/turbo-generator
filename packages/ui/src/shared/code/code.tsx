"use client";

import type { JSX, ReactNode } from "react";

interface CodeProps {
  children: ReactNode;
  className?: string;
}

const Code = ({ children, className }: CodeProps): JSX.Element => {
  return <code className={className}>{children}</code>;
};

export { type CodeProps, Code };
