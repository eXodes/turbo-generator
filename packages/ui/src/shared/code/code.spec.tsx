import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

import { Code } from "./code";

describe("Code", () => {
  it("should have 'Code' page title", async () => {
    const { baseElement } = render(<Code>Code</Code>);

    expect(baseElement).toMatchSnapshot();
  });
});
