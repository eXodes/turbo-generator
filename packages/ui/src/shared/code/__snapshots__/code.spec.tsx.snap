// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Code > should have 'Code' page title 1`] = `
<body>
  <div>
    <code>
      Code
    </code>
  </div>
</body>
`;
